sources/models/src/modeled_types/kubernetes.rs

```diff
@@ -27,10 +28,10 @@ lazy_static! {
     pub(crate) static ref KUBERNETES_NAME: Regex = Regex::new(r"^[0-9a-z.-]{1,253}$").unwrap();
 }
 
-impl TryFrom<&str> for KubernetesName {
-    type Error = error::Error;
+impl FromStr for KubernetesName {
+    type Err = error::Error;
 
-    fn try_from(input: &str) -> Result<Self, Self::Error> {
+    fn from_str(input: &str) -> Result<Self, Self::Err> {
         ensure!(
```

Comment:
I could be misunderstanding here, but, since we have declared `KubernetesName` as public, is it reasonable to remove `TryFrom`? Could we not just keep it, while adding `FromStr` as well? This same question is valid below as well with `KubernetesLabelKey`

Reply:
It feels a bit odd and misleading to have both.  Is there any reason to remove `TryFrom` here other than consistency?

Reply:
I would be more concerned if this was a public library used in multiple places since this change will actually cause pain to the users and require refactoring. But, since we are the only consumers of this code, and the painful refactor was already done, it should be fine to only keep `FromStr`.

Downstream forks of Bottlerocket may have to do some refactor as well if they rely on what `TryFrom` provides, but that is also true for whatever new feature we introduce in Bottlerocket, specially in the API model.